### ISSUE TITLE
Support Transit writer options

### ISSUE THREAD

**Issue by cjhowe7** [juxt/yada#193]
When creating a transit reader or writer in Clojure, there is a third argument which is a map of options. It would be nice if yada allowed specifying those options, somehow. This allows custom handlers for data types, like `om.next`'s tempids. I think extensibility is a big feature of transit, so it'd make sense to support this extensibility somehow.

Right now, you have to copy the `yada.transit` namespace, but since it uses things outside of `yada.yada` those APIs may change. So this solution is less than ideal.

I'm not sure exactly what changes would be required to do this. Here is one place where the options argument would need to be passed:

https://github.com/juxt/yada/blob/432d1b25f83a4af5d94d1108f37ee253a2a74bce/ext/transit/src/yada/transit.clj#L14

**Comment by malcolmsparks** [comment]
Agreed. This is a bit tricky and calls for an improved `yada.body.MessageBody` protocol which gives access to the request-context (and resource metadata), because you need some way of communicating some resource configuration to transit.
